front/src/modules/companies/hooks/useOpenActionBar.tsx

```diff
@@ -0,0 +1,81 @@
+import { getOperationName } from '@apollo/client/utilities';
+import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil';
+
+import { useOpenCreateActivityDrawerForSelectedRowIds } from '@/activities/hooks/useOpenCreateActivityDrawerForSelectedRowIds';
+import { GET_PIPELINES } from '@/pipeline/queries';
+import { ActionBarEntry } from '@/ui/action-bar/components/ActionBarEntry';
+import { IconCheckbox, IconNotes, IconTrash } from '@/ui/icon';
+import { useResetTableRowSelection } from '@/ui/table/hooks/useResetTableRowSelection';
+import { actionBarEntriesState } from '@/ui/table/states/ActionBarEntriesState';
+import { selectedRowIdsSelector } from '@/ui/table/states/selectedRowIdsSelector';
+import { tableRowIdsState } from '@/ui/table/states/tableRowIdsState';
+import {
+  ActivityType,
+  CommentableType,
+  useDeleteManyCompaniesMutation,
+} from '~/generated/graphql';
+
+export function useOpenActionBar() {
+  const setActionBarEntries = useSetRecoilState(actionBarEntriesState);
+
+  const openCreateActivityRightDrawer =
+    useOpenCreateActivityDrawerForSelectedRowIds();
+
+  async function handleActivityClick(type: ActivityType) {
+    openCreateActivityRightDrawer(type, CommentableType.Company);
+  }
+
+  const selectedRowIds = useRecoilValue(selectedRowIdsSelector);
+
+  const resetRowSelection = useResetTableRowSelection();
+
+  const [deleteCompanies] = useDeleteManyCompaniesMutation({
+    refetchQueries: [getOperationName(GET_PIPELINES) ?? ''],
+  });
+
+  const [tableRowIds, setTableRowIds] = useRecoilState(tableRowIdsState);
+
+  async function handleDeleteClick() {
+    const rowIdsToDelete = selectedRowIds;
+
+    resetRowSelection();
+
+    await deleteCompanies({
```

Comment:
we should extract this to a company hook (for example useDeleteCompanyHook) as this seems duplicate code between ActionBar and ContextMenu